### ISSUE TITLE
@urql/exchange-persisted-fetch doesn't persist the query

### ISSUE THREAD

**Issue by TLadd** [FormidableLabs/urql#764]
urql version & exchanges:
urql: 1.9.7
@urql/exchange-persisted-fetch: 0.1.0

**Steps to reproduce**

1. Using this urql client config. Note that I see the same behavior regardless of whether I'm setting `preferGetMethod: true` or not.
```
const client = createClient({
  url: "http://localhost:3001/graphql",
  exchanges: [
    dedupExchange,
    cacheExchange,
    persistedFetchExchange,
    fetchExchange,
  ],
  preferGetMethod: true,
});
``` 
2. Load app and run against an Apollo Server. Observe that queries are never actually persisted.

Every page load, it sends an initial request with the following query params:
```
variables: {"where":{"url":"http://localhost:3000/"}}
extensions: {"persistedQuery":{"sha256Hash":"261d8cf2da51840aad856147ca6af3e8b358b5d81b509a50292112df4e7347a3","version":1}}
```
This is correct. The next request, however, has the query parameters

```
query: query GetPageBookmarkCount($where: PageWhereUniqueInput!) {
  pageBookmarkCount(where: $where)
}
variables: {"where":{"url":"http://localhost:3000/"}}
```

which correctly executes the query, but doesn't result in the query being persisted to the hash.

In the [apollo docs](https://www.apollographql.com/docs/apollo-server/performance/apq/#verify), there's an intermediary second step where the query gets stored the cache: `curl -g 'http://localhost:4000/?query={__typename}&extensions={"persistedQuery":{"version":1,"sha256Hash":"ecf4edb46db40b5132295c0291d62fb65d6759a9eedfa4d5d612dd5ec54a6b38"}}'` that doesn't seem to be happening.

**Expected behavior**

Persisted queries get saved to cache after the first request

**Actual behavior**

Persisted queries never get saved to cache

Here is a codesandbox that also exhibits this behavior: https://codesandbox.io/s/bold-field-l2d2v

**Comment by JoviDeCroock** [comment]
It sais `{"errors":[{"message":"PersistedQueryNotFound",` this means that the exchange will fallback to normal queries instead. Maybe this is turned off on your reproduction server

**Comment by kitten** [comment]
I suppose the problem here was that a spec doesn’t exist, so we haven’t done a lot of testing (hence it being a `0.1.0`)

I think I’ll spin up a persisted queries testing API so we can validate a fix to the library correctly 👍

**Comment by kitten** [comment]
I finally found the expected behaviour down here in the readme: https://github.com/apollographql/apollo-link-persisted-queries?files=1#protocol

This behaviour is defined in "Missing hash path", Step 4:

> Client sends both hash and query string to Server

**Comment by kitten** [comment]
The fix has been released in [@urql/exchange-persisted-fetch@0.1.1](https://github.com/FormidableLabs/urql/releases/tag/%40urql%2Fexchange-persisted-fetch%400.1.1)
